MdeModulePkg/Universal/HiiDatabaseDxe/ConfigRouting.c

```diff
@@ -3403,11 +3431,23 @@ GetBlockElement (
   while ((Link != &RequestBlockArray->Entry) && (Link->ForwardLink != &RequestBlockArray->Entry)) {
     BlockData     = BASE_CR (Link, IFR_BLOCK_DATA, Entry);
     NextBlockData = BASE_CR (Link->ForwardLink, IFR_BLOCK_DATA, Entry);
-    if ((NextBlockData->Offset >= BlockData->Offset) || (NextBlockData->Offset <= (BlockData->Offset + BlockData->Width))) {
-      if ((NextBlockData->Offset + NextBlockData->Width) > (BlockData->Offset + BlockData->Width)) {
-        BlockData->Width = (UINT16)(NextBlockData->Offset + NextBlockData->Width - BlockData->Offset);
+
+    // MU_CHANGE [BEGIN] - Use SafeIntLib functions

```

Comment:
All the same comments as above, most importantly that we can't continue in the fail cases because we need to hit the update step on line 3456 or else we will loop endlessly.

Reply:
Refactored - please double check

Reply:
LGTM